src/py/reactpy/reactpy/core/state_recovery.py

```diff
@@ -0,0 +1,295 @@
+import asyncio
+import base64
+from dataclasses import asdict, is_dataclass
+import datetime
+import hashlib
+import time
+from collections.abc import Iterable
+from decimal import Decimal
+from logging import getLogger
+from pathlib import Path
+from typing import Any, Callable
+from uuid import UUID
+
+import orjson
+import pyotp
+from more_itertools import chunked
+
+logger = getLogger(__name__)
+
+
+class StateRecoveryFailureError(Exception):
+    """
+    Raised when state recovery fails.
+    """
+
+
+class StateRecoveryManager:
+    def __init__(
+        self,
+        serializable_types: Iterable[type],
+        pepper: str,
+        otp_key: str | None = None,
```

Comment:
An OTP key. If one is not provided, it is generated by listing out the directories under which reactpy is installed and grabbing the creation time. The OTP values are added to the hash